book/src/proposals/abi-management.md

```diff
@@ -0,0 +1,434 @@
+# Solana ABI management process
+
+This document proposes Solana ABI management process. The ABI management process
+is an engineering practice and supporting technical framework to avoid
+introducing unintended incompatible ABI changes.
+
+# Problem
+
+The Solana ABI (binary interface to the cluster) is currently only defined 
+implicitly by the implementation, and requires a very careful eye to notice
+breaking changes. This makes it extremely difficult to upgrade the software
+on an existing cluster without rebooting the ledger.
+
+# Requirements and objectives
+
+- Unintended ABI changes can be detected as CI failures mechanically.
+- Newer impl must be able to process oldest data (since genesis) once we go
+  mainnet. (This is in stark contrast with regard to conventional software;
```

Comment:
I hope this holds true for DLT... But might be wrong?

Reply:
I think it holds true.  IMO remove this entire block though: `(This is in ... replayable today)`, it's a little wordy and doesn't really help understanding